lib/Doctrine/Common/Cache/FileCache.php

```diff
@@ -126,9 +114,9 @@ protected function getFilename($id)
     {
         return $this->directory
             . DIRECTORY_SEPARATOR
-            . implode(str_split(hash('sha256', $id), 2), DIRECTORY_SEPARATOR)
+            . substr(hash('sha256', $id), 0, 2)
             . DIRECTORY_SEPARATOR
-            . preg_replace($this->disallowedCharacterPatterns, $this->replacementCharacters, $id)
+            . str_replace('.', '%2E', rawurlencode($id))
```

Comment:
As per discussion, needs encoding of uppercase ascii. Actually, to simplify, can't we have encoding of all chars?